### PR TITLE
Add an admin API for unprotecting local media from quarantine

### PR DESCRIPTION
**Pull request by dklimpel**
Related: #9086

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

File: docs/admin_api/media_admin_api.md [hunk 1]
```diff
@@ -159,6 +160,26 @@ Response:
...
 ```
 
+## Unprotecting media from being quarantined
+
+This API reverts the protection of a media.
+
+Request:
+
+```
+DELETE /_synapse/admin/v1/media/protect/<media_id>
```
Comment (richvdh) [inline docs/admin_api/media_admin_api.md:170]: does `DELETE` feel scary to anyone else? I feel like I'd be worrying that I'm deleting the media itself.

I'd be inclined towards `POST /_synapse/admin/v1/media/unprotect/<media_id>`, but I don't have a very strong opinion.

**Review by richvdh — COMMENTED** [review]
lgtm otherwise

File: docs/admin_api/media_admin_api.md (hunk 1 above)
Comment (dklimpel) [reply docs/admin_api/media_admin_api.md:170]: I do not like so many URI. When you something create with `PUT` or `PULL` you can revert it with `DELETE`. This ist HTTP.
We can add a sentence, that this do not delete a media. It only deletes the protection.
Do you have team call for discussing this? IMO there ist missin a kind of compliance for all oh the admin API.
Probably is need for an extensive API like for user.

Comment (clokep) [reply docs/admin_api/media_admin_api.md:170]: I agree that `DELETE` seems like the wrong semantic since you're updating the media, not deleting it.

Comment (richvdh) [reply docs/admin_api/media_admin_api.md:170]: @dklimpel so I did a straw poll among the team in #synapse-dev and the general feeling was that POST would be more intuitive than DELETE.

Although `DELETE` is in some ways the opposite of `POST`, I don't think that's always the case, and it's certainly not the case that there are any standards around that.

Comment (dklimpel) [reply docs/admin_api/media_admin_api.md]: Ok. Now it is a `POST` request.

File: changelog.d/10040.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Add an admin API for unprotecting local media from quarantine.
```
Comment (richvdh) [inline changelog.d/10040.feature:1]:
```suggestion
Add an admin API for unprotecting local media from quarantine. Contributed by @dklimpel.
```

**Review by richvdh — APPROVED** [review]
thank you!

**Pull request merged**